### PR TITLE
redeem ix

### PR DESCRIPTION
**Pull request by sebatustra**
## Summary by CodeRabbit

* **New Features**
  * Users can now redeem vault shares to retrieve underlying assets with automated fee deductions, proper balance tracking, and comprehensive validation to ensure vault state integrity and prevent errors throughout the redemption process.

* **Tests**
  * Added integration tests for redemption functionality covering multiple fee scenarios, detailed account balance verification, and edge case error handling.

### CODE REVIEW

**Comment by sebatustra** [comment]
@coderabbitai review

**Comment by LHFVII** [comment]
Could we add a test case for redeem that takes into account the transfer fees extension?

File: programs/vault/src/instructions/redeem.rs [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    }
+
+    // transfer from vault to user
+    ctx.accounts.transfer_assets_to_user(user_assets_out)?;
+
+    // decrease vault assets
+    ctx.accounts.vault.decrease_asset_supply(total_assets_out)?;
+
+    Ok(())
+}
```
[the remark this answers is not in the document]
Comment (sebatustra) [reply programs/vault/src/instructions/redeem.rs:183]: for the moment it is acceptable to rely on implicit validations, as we are doing for other cases (burn will fail if user does not have the amount)

**Comment by LHFVII** [comment]
LGTM
